packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -224,13 +231,48 @@ class TabController extends ChangeNotifier {
           notifyListeners();
         });
     } else {
+      final bool wasIndexChanging = _indexIsChangingCount > 0;
       _indexIsChangingCount += 1;
+      // should only notify AnimationStartedListener when
+      // it was not called by ScrollEndNotification or
+      // index was changing before (so that TabBar and TabBarView will start a new animation with new target index.)
+      // This happens when controller.index = ? is called while animation was ongoing.
+      if (_animationController.value != _index.toDouble() || wasIndexChanging) {
```

Comment:
I remember last time i check `ScrollEndNotification` is called controller.index = ? with the same value as controller.index. If we have the check on line 216, it should not have enter this block. Then this check is not necessary.

Reply:
Same as above comment, we already had that check but still went all the way to notifying AnimationStartedListener, so we need this check.

Reply:
In that case we still need to notify the listener, if we think like a tabcontroller, there is no reason to block this because someone does ask us to change our index.

Either we fixes the tabview to not trigger this weird case or ignore this notify.

Reply:
Ok, if we think ```AnimationStartedListener``` as ```IndexChangeStartedListener```, I guess notifying the listener does make sense.